Lib/codeop.py

```diff
@@ -102,11 +102,18 @@ def _maybe_compile(compiler, source, filename, symbol):
     try:
         if code:
             return code
-        if not code1 and repr(err1) == repr(err2):
+        if not code1 and _is_syntax_error(err1, err2):
             raise err1
     finally:
         err1 = err2 = None
 
+def _is_syntax_error(err1, err2):
+    if all("was never closed" in repr(err) for err in (err1, err2)):
+        return False
+    if repr(err1) == repr(err2):
```

Comment:
A bit hacky but it works for now.  I prefer the following, which is likely faster and to me easier to read.
```suggestion
    rep1 = repr(err1)
    rep2 = repr(err2)
    if "was never closed" in rep1 and"was never closed" in rep2:
        return False
    if rep1 == rep2:
```

Reply:
I agree is a bit hacky, but the way codeop is determining to continue or not is quite more hacky as is determining that based on the fact that the exceptions are different (as now both raise the same error message we need to do this). The alternative would be to include something that makes the exceptions different, which is even more hacky